### DISCUSSION TITLE
Storefront API - Is there a way to schedule the publication of a product?

### DISCUSSION THREAD

**Discussion by Stalex89** [Shopify/storefront-api-feedback#155] [Help]
Hello, 

Sorry if posting a duplicate topic, but couldn't find similar in this repo or other forums while googling.

In Shopify, there is an option to [set future availability of the product](https://help.shopify.com/en/manual/shopify-admin/productivity-tools/future-publishing) in online store, but as far as I see this information is not available in the Storefront API (at least I cannot find it in the docs or forums). 

Is there maybe an implementation of this feature? or am I missing something ?
Thank you in advance!

**Comment by johncraigcole** [comment]
Hey @Stalex89 👋. You can't publish to an app channel other than online store with a date in the future.  

For a Storefront API app channel used with a custom storefront / headless site, it's currently only possible to publish on demand with the Admin API using the [publishablePublish mutation](https://shopify.dev/api/admin-graphql/2022-07/mutations/publishablePublish) by setting a publish date to a date and time in the past. You can get the publication id of your Storefront API app channel with a [publications query](https://shopify.dev/api/admin-graphql/2022-07/queries/publications) which again is part of the Shopify Admin API.
